lib/hsa/mcwamp_hsa.cpp

```diff
@@ -4454,30 +4454,61 @@ hsa_status_t HSACopy::hcc_memory_async_copy(Kalmar::hcCommandKind copyKind, cons
 
     hsa_agent_t hostAgent = const_cast<Kalmar::HSADevice *> (copyDeviceArg)->getHostAgent();
 
-    void *dstPtr = dstPtrInfo._devicePointer;
-    void *srcPtr = srcPtrInfo._devicePointer;
+    /* Determine src and dst pointer passed to ROCR runtime.
+     *
+     * Pre-condition:
+     * - this->dst and this->src must be tracked by AM API implemantation.
+     * - dstPtrInfo and srcPtrInfo must be valid.
+     */
+    void *dstPtr = nullptr;
+    void *srcPtr = nullptr;
 
     hsa_agent_t srcAgent, dstAgent;
     switch (copyKind) {
         case Kalmar::hcMemcpyHostToHost: 
             srcAgent=hostAgent; dstAgent=hostAgent;
 
-            // Use host pointers since this copy will be performed with CPU.
-            // If pointers are registered, then devicePointer may not match host pointer.
```

Comment:
I think it is important to retain the comment that we are expecting ROCR runtime to continue to use host pointers for host to host copies.  (rather than using GPU copy engine which would use device pointers)